lib/server.rb

```diff
@@ -1,45 +1,56 @@
-require 'socket'
 require 'pry'
+require 'socket'
 
 class Server
-  attr_reader :tcp_server
+  attr_reader :tcp_server, :client, :count
 
-  def initialize(port = 9292)
-    @tcp_server = TCPServer.new(port)
+  def initialize
+    @tcp_server = TCPServer.new(9292)
+    @client = nil
     @count = 0
   end
 
-  def send_response(client, args)
-      client.puts args[:output]
+  def send_response(headers, output)
+    client.puts headers
+    client.puts output
+    client.close
+    @count += 1
+  end
 
-    # if var == quit
-    # client.close
-    # else
-    # accept
-    # end
+  def reset_count
+    @count = -1
   end
 
-  def build_response_message(client)
-    response = "<p>Hello, World (0)</p>"
+  def format_response(request_lines)
+    reset_count if request_lines[0].include? "/clear_count"
+
+    response = "<p>Hello, World! (#{count})</p>"
     output = "<html><head></head><body>#{response}</body></html>"
-    send_response(client, output: output)
+    headers = ["http/1.1 200 ok",
+               "date: #{Time.now.strftime('%a, %e %b %Y %H:%M:%S %z')}",
```

Comment:
Do not use `Time.now.strftime` without zone. Use one of `Time.zone.now.strftime`, `Time.now.strftime.current`, `Time.now.strftime.in_time_zone`, `Time.now.strftime.utc`, `Time.now.strftime.getlocal`, `Time.now.strftime.iso8601`, `Time.now.strftime.jisx0301`, `Time.now.strftime.rfc3339`, `Time.now.strftime.to_i`, `Time.now.strftime.to_f` instead.